### PR TITLE
Feat/archive filters

### PR DESCRIPTION
**Pull request by smaraghi**
Archive Filters

- Styles have been modified with approval of Christina - new mock found on Figma:
https://www.figma.com/file/cAg0A1ww0s2KbJdIbUvl2j/PONI-Redesign-Handoff

- New version of facet JS// No Jquery
- All Facet CSS unset
- Checkbox Filters: The HTML for the fSelect checkboxes and the regular checkboxes are different so they were styled separately
- Dropdowns turned into MultiSelects
- Cannot bold Author and unbold Author name
- Filter functionality: As you select one filter, it filters down the others. Filter upon selection

Note: There is no data currently. Mock data has been added into the development branch.

### CODE REVIEW

File: wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      opacity: 0.4;
+    }
+
+    &.d1 .fs-option-label {
+      padding-left: 0 !important;
+    }
+
+    &.d1 .fs-checkbox {
+      padding-left: rem(20px);
+    }
```
Comment (urchykli) [inline wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss:278]: I wasn't able to find anything with a class of `d1`, but please let me know if I'm missing something.

File: wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      white-space: normal !important;
+      word-break: break-word;
+    }
+  }
+
+  &-optgroup-label {
+    font-weight: bold;
+    text-align: center;
+    background-color: #f8f8f8;
+  }
```
Comment (urchykli) [inline wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss:290]: I wasn't able to find this class either.

**Review by urchykli — CHANGES_REQUESTED** [review]
Thank you so much for jumping in on this, Serven! This was a great first pass and the filters look great.  I just had a few comments below and some minor things inline.
- [x] The clickable area for the checkboxes extends the full length of the `div`, but it should stop after the text.
- [x] Pagination/# of results still show total for all posts after checkbox is selected.
- [x] We should ask Christina how the focus should look on the checkboxes.  Right now it's an outline around the checkbox name and the checkbox isn't included.
- [x] There are two other variations of the filter, one on the search/tag/topic archives and one on the author archive.
- [x] When a checkbox is checked or something is selected in a dropdown, the dropdowns flash blue.

File: wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+    .fs-default & {
+      color: $color-black-100;
+    }
+  }
+
+  .fs-label {
+    @extend %text-style-label-large;
+    color: $color-black-190;
+  }
```
[the remark this answers is not in the document]
Comment (jnschrag) [reply wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss:168]: Yes, this controls the default label of "All Authors", "All Series", etc.

**Comment by jnschrag** [comment]
@urchykli This is ready for you to review again. Apologies in advance, it's **a lot**.

You'll want to pull down from the Dev environment, as I did a couple of things there:
- I created a new taxonomy `filterable_content_type` that is attached to Posts, Events, and Program Updates. This is what the "Analysis Subtype" taxonomy should have been. Once you take a look, we should delete the Analysis Subtype taxonomy to avoid confusion.
  - I went through and tagged all Analysis posts as "Analysis --> Commentary" under the content type taxonomy. I also went and tagged the 5 reports as "Reports" instead of commentary.
  - I added the first page of events as "Event"
  - I added "Program Update" to all of the "Program Updates" posts so they should be organized correctly
- I added & modified some of the facets, included added 2 that handle pagination results and the numbers/navigation

## Other Notes
- This removes the Search & Filter Pro plugin. So there's a lot of files included in this PR that are just deletions. Sorry about that! Had to get rid of it because it was conflicting with the FacetWP stuff.
- I did the best I could to account for the different use cases for the filters. We'll likely need to make some modifications to `archive.php` as we start building out the individual archives though.
- I made some modifications to the existing pagination functions so they match up with the FacetWP pagination.
- There's probably going to be some continued funkiness with the pagination based on the featured posts thing. I'm happy to take a look at that if needed after the rest of this PR is accepted & merged in.

File: wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+  &-facet:last-of-type {
+    margin-bottom: rem(40) !important;
+  }
+
+  /*----------  Facet: checkboxes  ----------*/
+
+  &-checkbox {
+    display: flex;
+    align-items: top;
```
Comment (urchykli) [inline wp-content/themes/csisnuclearnetwork/assets/_scss/components/_facet.scss:23]: This is an invalid property.

**Review by urchykli — CHANGES_REQUESTED** [review]
Thank you for jumping in on this, Jacque!  The filters look great!

Other than what we discussed with the subtypes, the filters should be two columns on the medium breakpoint and the author archive shouldn't have the author dropdown.

**Review by urchykli — APPROVED** [review]

**Pull request merged**